script/check-size

```diff
@@ -65,6 +65,13 @@ markdown_init()
     echo '| :----: | :------: | -----: | ----: | ---: | ----: |'
 }
 
+nm_size()
+{
+    arm-none-eabi-nm --print-size --size-sort -C "$1" > nmsize_old
+    arm-none-eabi-nm --print-size --size-sort -C "$2" > nmsize_new
+    diff -Nuar nmsize_old nmsize_new
+}
+
```

Comment:
This is not directly related to rest of changes, I am just curious. (I am fine with it being added in this PR).

Reply:
I added this to check what caused the code size increase. Still working on it.

Reply:
The size increase is mainly caused by the newly added `CliUartOutput`(in examples/apps/cli/cli_uart.cpp) which includes an enhancement for output long string. It eliminate a big buffer in the stack for formatting output string by reusing the whole Uart tx buffer for this purpose. The cost is text size increase. I think it acceptable.

Reply:
removed from this PR.